pkg/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go

```diff
@@ -109,35 +137,33 @@ func (k *bootstrapController) sync(ctx context.Context, controllerContext factor
 	}
 
 	// the bootstrap kubeconfig secret is changed, reload the klusterlet agents
-	return k.reloadAgents(ctx, controllerContext, klusterletNamespace, klusterletName)
+	reloadReason := fmt.Sprintf("the bootstrap secret %q is changed", klusterletNamespace+"/"+helpers.BootstrapHubKubeConfig)
+	return k.reloadAgents(ctx, controllerContext, klusterletNamespace, klusterletName, reloadReason)
 }
 
 // reloadAgents reload klusterlet agents by
 // 1. make the registration agent re-bootstrap by deleting the current hub kubeconfig secret to
 // 2. restart the registration and work agents to reload the new hub ca by deleting the agent deployments
-func (k *bootstrapController) reloadAgents(ctx context.Context, ctrlContext factory.SyncContext, namespace, klusterletName string) error {
+func (k *bootstrapController) reloadAgents(ctx context.Context, ctrlContext factory.SyncContext, namespace, klusterletName, reason string) error {
 	if err := k.kubeClient.CoreV1().Secrets(namespace).Delete(ctx, helpers.HubKubeConfig, metav1.DeleteOptions{}); err != nil {
 		return err
 	}
-	ctrlContext.Recorder().Eventf("HubKubeconfigSecretDeleted",
-		fmt.Sprintf("the hub kubeconfig secret %q is deleted due to the bootstrap secret %q is changed",
-			namespace+"/"+helpers.HubKubeConfig, namespace+"/"+helpers.BootstrapHubKubeConfig))
+	ctrlContext.Recorder().Eventf("HubKubeconfigSecretDeleted", fmt.Sprintf("the hub kubeconfig secret %q is deleted due to %s",
+		namespace+"/"+helpers.HubKubeConfig, reason))
 
 	registrationName := fmt.Sprintf("%s-registration-agent", klusterletName)
 	if err := k.kubeClient.AppsV1().Deployments(namespace).Delete(ctx, registrationName, metav1.DeleteOptions{}); err != nil {
 		return err
 	}
-	ctrlContext.Recorder().Eventf("KlusterletAgentDeploymentDeleted",
-		fmt.Sprintf("the deployment %q is deleted due to the bootstrap secret %q is changed",
-			namespace+"/"+registrationName, namespace+"/"+helpers.BootstrapHubKubeConfig))
+	ctrlContext.Recorder().Eventf("KlusterletAgentDeploymentDeleted", fmt.Sprintf("the deployment %q is deleted due to %s",
+		namespace+"/"+registrationName, reason))
 
 	workName := fmt.Sprintf("%s-work-agent", klusterletName)
 	if err := k.kubeClient.AppsV1().Deployments(namespace).Delete(ctx, workName, metav1.DeleteOptions{}); err != nil {
 		return err
 	}
-	ctrlContext.Recorder().Eventf("KlusterletAgentDeploymentDeleted",
-		fmt.Sprintf("the deployment %q is deleted due to the bootstrap secret %q is changed",
-			namespace+"/"+workName, namespace+"/"+helpers.BootstrapHubKubeConfig))
+	ctrlContext.Recorder().Eventf("KlusterletAgentDeploymentDeleted", fmt.Sprintf("the deployment %q is deleted due to %s",
+		namespace+"/"+workName, reason))
```

Comment:
it should be 
fmt.Sprintf("the deployment %s/%s is deleted due to %s",		namespace, workName, reason) ?

Reply:
used `%q` to quote the namespace and work name, e.g. `the deployment "test/test" is deleted due to ...` or it will be `the deployment test/test is deleted due to ...` which style will be better?

Reply:
using `%s/%s`